renderer/src/pages/Dashboard.tsx

```diff
@@ -0,0 +1,127 @@
+import { useEffect, useState, useCallback } from 'react'
+import {
+  getAllActivities, stopSaturnNode, startSaturnNode,
+  setFilAddress, getFilAddress,
+  getTotalEarnings, getTotalJobsCompleted
+} from '../lib/station-config'
+import { ActivityEventMessage } from '../typings'
+import ActivityLog from '../components/ActivityLog'
+import HeaderBackgroundImage from '../assets/img/header.png'
+import WalletIcon from '../assets/img/wallet.svg'
+import { useNavigate } from 'react-router-dom'
+
+const Dashboard = (): JSX.Element => {
+  const navigate = useNavigate()
+
+  const [address, setAddress] = useState<string | undefined>()
+  const [totalJobs, setTotalJobs] = useState<number>(0)
+  const [totalEarnings, setTotalEarnigs] = useState<number>(0)
+  const [activities, setActivities] = useState<ActivityEventMessage[]>([])
+  const shortAddress = (str: string | undefined) => str
+    ? str.substring(0, 4) + '...' + str.substring(str.length - 4, str.length)
+    : ''
+  const disconnect = async () => {
+    // TODO: move disconnect logic to backend
+    await stopSaturnNode()
+    await setFilAddress('')
+    setAddress(undefined)
+    navigate('/wallet', { replace: true })
+  }
+
+  const reload = async (): Promise<void> => {
+    setAddress(await getFilAddress())
+    setActivities(await getAllActivities())
+    setTotalEarnigs(await getTotalEarnings())
+    setTotalJobs(await getTotalJobsCompleted())
+  }
+
+  const handleActivity = useCallback((value: ActivityEventMessage[]) => {
+    setActivities(value)
+  }, [])
+
+  const handleEarnings = useCallback((value: number) => {
+    setTotalEarnigs(value)
+  }, [])
+
+  const handleJobs = useCallback((value: number) => {
```

Comment:
these shouldn't be necessary, you can just pass `setTotalJobs` directly, right?

Reply:
Currently yes, but I'd prefer to explicitly define a callback to handle such recalls. One in the future may want to process it other way or trigger anything else when this event is triggered.
Anyway, If you prefer the implicit way, I can change

Reply:
> One in the future may want to process it other way or trigger anything else when this event is triggered.

If that is the case then it's easy to update. We can keep the `useCallbacks` while this PR is still WIP, so you can iterate quickly should you need to change these, but let's please clean these up when the PR is done

Reply:
[afd5e98](https://github.com/filecoin-project/filecoin-station/pull/144/commits/afd5e98b0223258d90a6658c64dd3aea2af46f8a)